lms/djangoapps/letstudy_api/views.py

```diff
@@ -37,18 +42,22 @@ def post(self, request):
         data = request.data
         email = data.get('email')
         username = data.get('username')
-        uid = data.get('uid')
+        uid = data.get('uid', None)
         first_name = data.get('first_name', '')
         last_name = data.get('last_name', '')
-        full_name = "{} {}".format(first_name, last_name).strip() if first_name or last_name else username
+        full_name = "{} {}".format(first_name, last_name).strip() or username
 
-        if check_account_exists(username=username, email=email):
+        response_status = status.HTTP_400_BAD_REQUEST
+        if uid is None or UserSocialAuth.objects.filter(uid=uid).exists():
```

Comment:
I would create a complex error message if in the request more than one parameter is incorrect.